With the same idea of Q1 above using insertion sort and quick sort,  answer the following questions [total 15 points]:

a) How should we choose k in practice?  (Hint:  can you do any better than actual simulation, trial-and-errors?)  [0 points]

b) Find and fully understand the codes of quicksort and insertion sort.  Say "Yes" if you do.  [2 points]

c) Write separate test programs of both algorithms (insertion sort and quicksort) with a test data set of your choice and show they work. (perhaps the size of 20 integers) [3 points]

d) [10 points] Write a code combining both insertion sort and quicksort algorithms as described in Q1) to find out the most optimal k minimizing time efficiency of your algorithms using 1 M data set from HW2.  You may use the time efficiency program you wrote in the previous homework.  Please collect and display some performance data which proves your k is the most optimal as shown below:  

k-10: 11.42 seconds

k-5 :  11.07 seconds

k: 10.55 seconds

k+5: 10.59 seconds.

k+10: 11.02 seconds

Please note that the efficiency data displayed above are fictitious ones made up by me for demo purposes only.  They are not actual simulation results.

# B

# Quick Sort

In quick sort we first partition the array such that elements to the left of the pivot element are smaller, and elements on the right side of pivot are greater than pivot. Procedure is repeated for left and right subarray.Using randomly generated pivot can help us improve the time complexity of quick sort.
There are 2 methods to partition the array:
1. Hoare's partition
2. Lomuto's partition

In [ ]:
import random
def quickSort(arr,l,h):
    if(l<h):
        pivotindex=randpartition(arr,l,h)
        #array is now sorted around this pivot element
        #we will sort left and right half of the array
        quickSort(arr,l,pivotindex-1)
        quickSort(arr,pivotindex+1,h)
        
def randpartition(arr,l,h):
    randpivot= random.randrange(l,h)
    arr[l],arr[randpivot] = arr[randpivot],arr[l]
    return partition(arr,l,h)
        
#this next function takes the first element aas pivot ,places it at correct postion
#abd all other elements are arranged accordingly to this pviot i.e smaller on left side
#larger one on right side
def partition(arr,l,h):
    pivot= l #pivot 
    i= l+1 #to keep track where the partition is starting from
    for j in range(l+1,h+1):
        if arr[j]<=arr[pivot]:
            arr[i],arr[j] =arr[j],arr[i]
            i=i+1
    arr[pivot],arr[i-1] = arr[i-1],arr[pivot]
        
    pivot=i-1
    

In [ ]:
if __name__ == "__main__":
    array = [10, 7, 8, 9, 1, 5]
    quickSort(array, 0, len(array) - 1)
    print(array)

# Python implementation QuickSort using 
# Lomuto's partition Scheme.

In [ ]:
#https://www.geeksforgeeks.org/quicksort-using-random-pivoting/

import random
 
'''
The function which implements QuickSort.
arr :- array to be sorted.
start :- starting index of the array.
stop :- ending index of the array.
'''
def quicksort(arr, start , stop):
    if(start < stop):
         
        # pivotindex is the index where 
        # the pivot lies in the array
        pivotindex = partitionrand(arr,\
                             start, stop)
         
        # At this stage the array is 
        # partially sorted around the pivot. 
        # Separately sorting the 
        # left half of the array and the
        # right half of the array.
        quicksort(arr , start , pivotindex-1)
        quicksort(arr, pivotindex + 1, stop)
 
# This function generates random pivot,
# swaps the first element with the pivot 
# and calls the partition function.
def partitionrand(arr , start, stop):
 
    # Generating a random number between the 
    # starting index of the array and the
    # ending index of the array.
    randpivot = random.randrange(start, stop)
 
    # Swapping the starting element of
    # the array and the pivot
    arr[start], arr[randpivot] = \
        arr[randpivot], arr[start]
    return partition(arr, start, stop)
 
'''
This function takes the first element as pivot, 
places the pivot element at the correct position 
in the sorted array. All the elements are re-arranged 
according to the pivot, the elements smaller than the
pivot is places on the left and the elements
greater than the pivot is placed to the right of pivot.
'''
def partition(arr,start,stop):
    pivot = start # pivot
     
    # a variable to memorize where the 
    i = start + 1
     
    # partition in the array starts from.
    for j in range(start + 1, stop + 1):
         
        # if the current element is smaller
        # or equal to pivot, shift it to the
        # left side of the partition.
        if arr[j] <= arr[pivot]:
            arr[i] , arr[j] = arr[j] , arr[i]
            i = i + 1
    arr[pivot] , arr[i - 1] =\
            arr[i - 1] , arr[pivot]
    pivot = i - 1
    return (pivot)
 
# Driver Code
if __name__ == "__main__":
    array = [10, 7, 8, 9, 1, 5]
    quicksort(array, 0, len(array) - 1)
    print(array)
 


# Insertion sort  from homework2

In [ ]:
def insertionSort(score_ints):
    comparisions =0
    for i in range(1,len(score_ints)):
        value=score_ints[i]
        j=i-1
        while j>=0:
            if value<score_ints[j]:
                flag=True
            else:
                flag=False
            comparisions +=1
            if flag:
                score_ints[j+1] = score_ints[j]
                score_ints[j] = value
                j=j - 1
            else:
                break
    print("Number of comparisions :", comparisions)
    print("Sorted list:", score_ints)
                

# C

# Write separate test programs of both algorithms (insertion sort and quicksort) with a test data set of your choice and show they work. (perhaps the size of 20 integers)

In [89]:
import random
 
'''
The function which implements QuickSort.
arr :- array to be sorted.
start :- starting index of the array.
stop :- ending index of the array.
'''
def quicksort(arr, start , stop):
    if(start < stop):
         
        # pivotindex is the index where 
        # the pivot lies in the array
        pivotindex = partitionrand(arr,\
                             start, stop)
         
        # At this stage the array is 
        # partially sorted around the pivot. 
        # Separately sorting the 
        # left half of the array and the
        # right half of the array.
        quicksort(arr , start , pivotindex-1)
        quicksort(arr, pivotindex + 1, stop)
 
# This function generates random pivot,
# swaps the first element with the pivot 
# and calls the partition function.
def partitionrand(arr , start, stop):
 
    # Generating a random number between the 
    # starting index of the array and the
    # ending index of the array.
    randpivot = random.randrange(start, stop)
 
    # Swapping the starting element of
    # the array and the pivot
    arr[start], arr[randpivot] = \
        arr[randpivot], arr[start]
    return partition(arr, start, stop)
 
'''
This function takes the first element as pivot, 
places the pivot element at the correct position 
in the sorted array. All the elements are re-arranged 
according to the pivot, the elements smaller than the
pivot is places on the left and the elements
greater than the pivot is placed to the right of pivot.
'''
def partition(arr,start,stop):
    pivot = start # pivot
     
    # a variable to memorize where the 
    i = start + 1
     
    # partition in the array starts from.
    for j in range(start + 1, stop + 1):
         
        # if the current element is smaller
        # or equal to pivot, shift it to the
        # left side of the partition.
        if arr[j] <= arr[pivot]:
            arr[i] , arr[j] = arr[j] , arr[i]
            i = i + 1
    arr[pivot] , arr[i - 1] =\
            arr[i - 1] , arr[pivot]
    pivot = i - 1
    return (pivot)

In [8]:
 
# Driver Code
if __name__ == "__main__":
    array = [5,8,1,19,21,94,36,45,30,16,75,62,85,47,12,99,55,39,83,10]
    quicksort(array, 0, len(array) - 1)
    print(array)

[1, 5, 8, 10, 12, 16, 19, 21, 30, 36, 39, 45, 47, 55, 62, 75, 83, 85, 94, 99]


In [9]:
score_ints=[5,8,1,19,21,94,36,45,30,16,75,62,85,47,12,99,55,39,83,10]

In [10]:
def insertionSort(score_ints):
    comparisions =0
    for i in range(1,len(score_ints)):
        value=score_ints[i]
        j=i-1
        while j>=0:
            if value<score_ints[j]:
                flag=True
            else:
                flag=False
            comparisions +=1
            if flag:
                score_ints[j+1] = score_ints[j]
                score_ints[j] = value
                j=j - 1
            else:
                break
    print("Number of comparisions :", comparisions)
    print("Sorted list:", score_ints)
                

In [11]:
insertionSort(score_ints)

Number of comparisions : 82
Sorted list: [1, 5, 8, 10, 12, 16, 19, 21, 30, 36, 39, 45, 47, 55, 62, 75, 83, 85, 94, 99]


# D

Write a code combining both insertion sort and quicksort algorithms as described in Q1) to find out the most optimal k minimizing time efficiency of your algorithms using 1 M data set from HW2.  You may use the time efficiency program you wrote in the previous homework.  Please collect and display some performance data which proves your k is the most optimal as shown below:  

k-10: 11.42 seconds

k-5 :  11.07 seconds

k: 10.55 seconds

k+5: 10.59 seconds.

k+10: 11.02 seconds

In [83]:
#modified code from geekforgeeks

import random
import time
 
 # The following two functions are used  
# to perform quicksort on the array.  
  
# Partition function for quicksort 
def partition(numList, first, last):
    piv = numList[last]
    i = first-1
    for j in range(first,last):
        if numList[j] < piv:
            i += 1
            temp = numList[i]
            numList[i] = numList[j]
            numList[j] = temp
 
    tempo = numList[i+1]
    numList[i+1] = numList[last]
    numList[last] = tempo
 
    return i+1
 
def insert_sort(numList, first, last):
    for x in range(first, last+1):
        key = numList[x]
        y = x-1
        while y > -1 and numList[y]> key:
            numList[y+1] = numList[y]
            y = y-1
        numList[y+1] = key
 # Hybrid function -> Quick + Insertion sort 
def hybrid_sort(arr , low , high , k):
    while(low<high):
          
        # If the size of the array is less  
        # than threshold apply insertion sort  
        # and stop recursion 
        if(high - low+1 < k):
            insert_sort(arr , low, high)
            break
        else :
            # Optimised quicksort which works on  
            # the smaller arrays first 
  
            # If the left side of the pivot  
            # is less than right, sort left part 
            # and move to the right part of the array 
            pivot = partition(arr , low , high)
            if(pivot - low < high - pivot):
                hybrid_sort(arr , low , pivot - 1 , k)
                low = pivot + 1
                 # If the right side of pivot is less  
                # than left, sort right side and  
                # move to the left side 
            else:
                hybrid_sort(arr , pivot +1 , high , k)
                high = pivot - 1
 
 

In [ ]:
with open("rand1000000.txt", "r") as f:
    score1 = f.read() # Read all file in case values are not on a single line
    score_ints1 = [ int(x) for x in score1.split() ] # Convert strings to ints
    #print(score_ints1) # sum all elements of the list

# section 5
# Run this  in Pycharm as it gives me size output error here, I have commented print statements becuase sorted list is too big to print. We can fix that by doing this in  jupyter notebook command line:
# Try this:

# jupyter notebook --NotebookApp.iopub_data_rate_limit=1.0e10
# Or this:

# yourTerminal:prompt> jupyter notebook --NotebookApp.iopub_data_rate_limit=1.0e10

In [78]:
 
if __name__ == '__main__':
    for i in range(10 , 15):
        numList = random.sample(range(1000000),1000000)
        # print(numList)
        start = time.time()
        hybrid_sort(numList ,  0 , len(numList) - 1 , i)
        end = time.time()
        # print(numList) #commented out because we are only concerned about time here but to check whether my sorting logic is corrrect please 
        #read steps above in section 5
        print ("Time taken: " + str(end - start)+ " value of i : " + str(i))
 
 
       
 

Time taken: 5.105936050415039 value of i : 10
Time taken: 5.105936050415039 value of i : 10
Time taken: 3.971768617630005 value of i : 11
Time taken: 3.971768617630005 value of i : 11
Time taken: 4.675080299377441 value of i : 12
Time taken: 4.675080299377441 value of i : 12
Time taken: 4.309580326080322 value of i : 13
Time taken: 4.309580326080322 value of i : 13
Time taken: 4.61623215675354 value of i : 14
Time taken: 4.61623215675354 value of i : 14


# K= 10-15

Time taken: 4.99388861656189 value of i : 10

Time taken: 4.90662145614624 value of i : 11

Time taken: 4.924279451370239 value of i : 12

Time taken: 4.648666143417358 value of i : 13

Time taken: 4.792725563049316 value of i : 14

# K= 20-25

In [80]:
if __name__ == '__main__':
    for i in range(25 , 30):
        numList = random.sample(range(1000000),1000000)
        # print(numList)
        start = time.time()
        hybrid_sort(numList ,  0 , len(numList) - 1 , i)
        end = time.time()
        # print(numList) #commented out because we are only concerned about time here but to check whether my sorting logic is corrrect please 
        #read steps above in section 5
        print ("Time taken: " + str(end - start)+ " value of i : " + str(i))
 
 

Time taken: 5.403136730194092 value of i : 25
Time taken: 4.68584680557251 value of i : 26
Time taken: 5.065555572509766 value of i : 27
Time taken: 4.805042266845703 value of i : 28
Time taken: 5.364426612854004 value of i : 29


Time taken: 5.403136730194092 value of i : 25

Time taken: 4.68584680557251 value of i : 26

Time taken: 5.065555572509766 value of i : 27

Time taken: 4.805042266845703 value of i : 28

Time taken: 5.364426612854004 value of i : 29

# K= 35-40

In [81]:
if __name__ == '__main__':
    for i in range(35 , 40):
        numList = random.sample(range(1000000),1000000)
        # print(numList)
        start = time.time()
        hybrid_sort(numList ,  0 , len(numList) - 1 , i)
        end = time.time()
        # print(numList) #commented out because we are only concerned about time here but to check whether my sorting logic is corrrect please 
        #read steps above in section 5
        print ("Time taken: " + str(end - start)+ " value of i : " + str(i))
 
 

Time taken: 4.874989986419678 value of i : 35
Time taken: 4.903511047363281 value of i : 36
Time taken: 4.664669752120972 value of i : 37
Time taken: 5.01728630065918 value of i : 38
Time taken: 4.900742292404175 value of i : 39


Time taken: 4.874989986419678 value of i : 35
        
Time taken: 4.903511047363281 value of i : 36
        
Time taken: 4.664669752120972 value of i : 37
        
Time taken: 5.01728630065918 value of i : 38
        
Time taken: 4.900742292404175 value of i : 39

# 50-60

In [82]:
if __name__ == '__main__':
    for i in range(50 , 60):
        numList = random.sample(range(1000000),1000000)
        # print(numList)
        start = time.time()
        hybrid_sort(numList ,  0 , len(numList) - 1 , i)
        end = time.time()
        # print(numList) #commented out because we are only concerned about time here but to check whether my sorting logic is corrrect please 
        #read steps above in section 5
        print ("Time taken: " + str(end - start)+ " value of i : " + str(i))
 

Time taken: 5.490853309631348 value of i : 50
Time taken: 5.1764914989471436 value of i : 51
Time taken: 5.2054948806762695 value of i : 52
Time taken: 5.039182424545288 value of i : 53
Time taken: 5.1453657150268555 value of i : 54
Time taken: 5.578489780426025 value of i : 55
Time taken: 5.429208755493164 value of i : 56
Time taken: 5.620328426361084 value of i : 57
Time taken: 5.941431045532227 value of i : 58
Time taken: 5.772552251815796 value of i : 59


Time taken: 5.490853309631348 value of i : 50

Time taken: 5.1764914989471436 value of i : 51

Time taken: 5.2054948806762695 value of i : 52

Time taken: 5.039182424545288 value of i : 53

Time taken: 5.1453657150268555 value of i : 54

Time taken: 5.578489780426025 value of i : 55

Time taken: 5.429208755493164 value of i : 56

Time taken: 5.620328426361084 value of i : 57

Time taken: 5.941431045532227 value of i : 58

Time taken: 5.772552251815796 value of i : 59

# 80-90

In [84]:
if __name__ == '__main__':
    for i in range(80 , 90):
        numList = random.sample(range(1000000),1000000)
        # print(numList)
        start = time.time()
        hybrid_sort(numList ,  0 , len(numList) - 1 , i)
        end = time.time()
        # print(numList) #commented out because we are only concerned about time here but to check whether my sorting logic is corrrect please 
        #read steps above in section 5
        print ("Time taken: " + str(end - start)+ " value of i : " + str(i))
 

Time taken: 5.825800180435181 value of i : 80
Time taken: 5.939219236373901 value of i : 81
Time taken: 6.377457857131958 value of i : 82
Time taken: 5.703835964202881 value of i : 83
Time taken: 6.620039939880371 value of i : 84
Time taken: 6.47749137878418 value of i : 85
Time taken: 5.978348731994629 value of i : 86
Time taken: 6.527425289154053 value of i : 87
Time taken: 7.131261110305786 value of i : 88
Time taken: 7.202315807342529 value of i : 89


Time taken: 5.825800180435181 value of i : 80
        
Time taken: 5.939219236373901 value of i : 81
        
Time taken: 6.377457857131958 value of i : 82
        
Time taken: 5.703835964202881 value of i : 83
        
Time taken: 6.620039939880371 value of i : 84
        
Time taken: 6.47749137878418 value of i : 85
        
Time taken: 5.978348731994629 value of i : 86
        
Time taken: 6.527425289154053 value of i : 87
        
Time taken: 7.131261110305786 value of i : 88
        
Time taken: 7.202315807342529 value of i : 89

# Observation:
As we can see that as the value of k is increasing avg time is also increasing.

# In the class, we talked about three types of recursion to reach a termination of recursions. 

a) ***Lists those 3 types with a high-level description and a specific algorithm falling into each type.  (3 points)***

**Constant Fraction** : Where we divide by something and following algorithms fall within those 
   1. Merge Sort
   2. Binary Search
   3. Matrix Multiplications


Example:

Merge sort on just one element takes constant time. When we have n > 1 elements, we break down the running time as follows.

**Divide**: The divide step just computes the middle of the subarray, which takes constant time. Thus, D(n) = Θ(1).

**Conquer**: We recursively solve two subproblems, each of size n/2, which contributes 2T(n/2) to the running time.

**Combine**: We have already noted that the MERGE procedure on an n-element subarray takes time Θ(n), and so C(n) = Θ(n).

When we add the functions D(n) and C(n) for the merge sort analysis, we are adding a function that is Θ(n) and a function that is Θ(1). This sum is a linear function of n, that is, Θ(n). Adding it to the 2T(n/2) term from the “conquer” step gives the recurrence for the worst-case running time T(n) of merge sort:

T(n) = Θ(1), if n = 1; T(n) = 2T(n/2) + Θ(n), if n > 1.

**Variable Fraction**: of original problem size
    quick sort
    kth element search

  

Example:
    

Quick-sort()
      choose a pivot
      
      move the smaller elements from left
      
      move the bigger elements from right
      
      quick-sort on the bigger half of the array

      if half is less then X
      
         only then do an insertion sort on the other half <- this is a tail recursion insertion sort 
         
      else
      
         quick sort on this half also

**Constant Linear Search** of original problem size  (smaller by 1 by one) constant linear reductions
    Factorial:F(n)=F(n-1)*n
    Multiplication:M(n,m)=M(n-1)+m

Example:
    Factorial can be calculated using following recursive formula.

  n! = n * (n-1)!
    
  n! = 1 if n = 0 or n = 1

# Extra Credit

In [103]:
from os.path import getsize
from stat import *
import os

In [104]:
#used following examples to get this result
#https://www.tutorialspoint.com/python/os_listdir.htm
#https://docs.python.org/3/library/stat.html
#os.listdir(path)

def FindMe(startDirectory,subString):
    dirs=os.listdir(startDirectory) #directory that needs to be explored.
    bytesCount = 0
    for i in dirs:  #going into directory
        pathname = os.path.join(startDirectory,i)
        mode = os.stat(pathname).st_mode
        if S_ISDIR(mode): #Return non-zero if the mode is from a directory.
            bytesCount += FindMe(pathname,subString)
        if S_ISREG(mode):  # is it a regular file?
            if subString not in i: #check for the substring if not then "conitue"
                continue
            bytesCount += getsize(pathname)
            print("Path of file visiting: ",pathname)
        
            print("Total Bytes Now : ",bytesCount) #adding bytes after each count
    return (bytesCount)

In [105]:
output1= FindMe(r"C:\Users\ravisher\Desktop\CSCI3916_HW2", '.js') #you can modify the directory part depending upon how muc 
# you want to narrow down the results

output1
        

Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\auth.js
Total Bytes Now :  615
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\auth_jwt.js
Total Bytes Now :  1213
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\db.js
Total Bytes Now :  3385
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\accepts\index.js
Total Bytes Now :  5252
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\accepts\package.json
Total Bytes Now :  7507
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\array-flatten\array-flatten.js
Total Bytes Now :  1195
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\array-flatten\package.json
Total Bytes Now :  2862
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\body-parser\index.js
Total Bytes Now :  2656
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\body-parser\lib\read.

Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\qs\dist\qs.js
Total Bytes Now :  24660
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\qs\lib\formats.js
Total Bytes Now :  387
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\qs\lib\index.js
Total Bytes Now :  598
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\qs\lib\parse.js
Total Bytes Now :  9090
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\qs\lib\stringify.js
Total Bytes Now :  17320
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\qs\lib\utils.js
Total Bytes Now :  23524
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\qs\package.json
Total Bytes Now :  50549
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\qs\test\index.js
Total Bytes Now :  81
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node

1408761

# Output

Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\auth.js
Total Bytes Now :  615
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\auth_jwt.js
Total Bytes Now :  1213
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\db.js
Total Bytes Now :  3385
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\accepts\index.js
Total Bytes Now :  5252
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\accepts\package.json
Total Bytes Now :  7507
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\array-flatten\array-flatten.js
Total Bytes Now :  1195
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\array-flatten\package.json
Total Bytes Now :  2862
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\body-parser\index.js
Total Bytes Now :  2656
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\body-parser\lib\read.js
Total Bytes Now :  3894
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\body-parser\lib\types\json.js
Total Bytes Now :  4918
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\body-parser\lib\types\raw.js
Total Bytes Now :  6802
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\body-parser\lib\types\text.js
Total Bytes Now :  9087
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\body-parser\lib\types\urlencoded.js
Total Bytes Now :  14884
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\body-parser\package.json
Total Bytes Now :  24093
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\buffer-equal-constant-time\index.js
Total Bytes Now :  1045
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\buffer-equal-constant-time\package.json
Total Bytes Now :  2663
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\buffer-equal-constant-time\test.js
Total Bytes Now :  3676
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\bytes\index.js
Total Bytes Now :  3469
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\bytes\package.json
Total Bytes Now :  5500
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\content-disposition\index.js
Total Bytes Now :  10594
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\content-disposition\package.json
Total Bytes Now :  12984
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\content-type\index.js
Total Bytes Now :  4809
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\content-type\package.json
Total Bytes Now :  6977
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\cookie\index.js
Total Bytes Now :  4037
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\cookie\package.json
Total Bytes Now :  6223
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\cookie-signature\index.js
Total Bytes Now :  1230
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\cookie-signature\package.json
Total Bytes Now :  2767
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\cors\lib\index.js
Total Bytes Now :  6623
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\cors\package.json
Total Bytes Now :  8452
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\crypto\package.json
Total Bytes Now :  1455
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\debug\component.json
Total Bytes Now :  321
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\debug\karma.conf.js
Total Bytes Now :  2057
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\debug\node.js
Total Bytes Now :  2097
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\debug\package.json
Total Bytes Now :  4324
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\debug\src\browser.js
Total Bytes Now :  4734
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\debug\src\debug.js
Total Bytes Now :  9128
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\debug\src\index.js
Total Bytes Now :  9391
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\debug\src\inspector-log.js
Total Bytes Now :  9764
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\debug\src\node.js
Total Bytes Now :  15779
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\depd\index.js
Total Bytes Now :  10669
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\depd\lib\browser\index.js
Total Bytes Now :  1512
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\depd\lib\compat\callsite-tostring.js
Total Bytes Now :  2229
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\depd\lib\compat\event-listener-count.js
Total Bytes Now :  2567
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\depd\lib\compat\index.js
Total Bytes Now :  3988
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\depd\package.json
Total Bytes Now :  18307
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\destroy\index.js
Total Bytes Now :  1043
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\destroy\package.json
Total Bytes Now :  2836
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\dotenv\config.js
Total Bytes Now :  189
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\dotenv\lib\cli-options.js
Total Bytes Now :  303
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\dotenv\lib\env-options.js
Total Bytes Now :  725
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\dotenv\lib\main.js
Total Bytes Now :  3726
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\dotenv\package.json
Total Bytes Now :  5839
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\dotenv\types\tsconfig.json
Total Bytes Now :  278
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\dotenv\types\tslint.json
Total Bytes Now :  381
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\ecdsa-sig-formatter\package.json
Total Bytes Now :  2155
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\ecdsa-sig-formatter\src\ecdsa-sig-formatter.js
Total Bytes Now :  5009
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\ecdsa-sig-formatter\src\param-bytes-for-alg.js
Total Bytes Now :  5465
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\ee-first\index.js
Total Bytes Now :  1684
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\ee-first\package.json
Total Bytes Now :  3486
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\encodeurl\index.js
Total Bytes Now :  1586
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\encodeurl\package.json
Total Bytes Now :  3725
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\escape-html\index.js
Total Bytes Now :  1362
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\escape-html\package.json
Total Bytes Now :  2830
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\etag\index.js
Total Bytes Now :  2479
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\etag\package.json
Total Bytes Now :  4770
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\express\index.js
Total Bytes Now :  224
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\express\lib\application.js
Total Bytes Now :  14270
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\express\lib\express.js
Total Bytes Now :  16679
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\express\lib\middleware\init.js
Total Bytes Now :  853
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\express\lib\middleware\query.js
Total Bytes Now :  1738
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\express\lib\request.js
Total Bytes Now :  30921
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\express\lib\response.js
Total Bytes Now :  58045
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\express\lib\router\index.js
Total Bytes Now :  14883
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\express\lib\router\layer.js
Total Bytes Now :  18179
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\express\lib\router\route.js
Total Bytes Now :  22328
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\express\lib\utils.js
Total Bytes Now :  86345
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\express\lib\view.js
Total Bytes Now :  89671
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\express\package.json
Total Bytes Now :  93878
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\finalhandler\index.js
Total Bytes Now :  6518
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\finalhandler\package.json
Total Bytes Now :  8914
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\forwarded\index.js
Total Bytes Now :  1360
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\forwarded\package.json
Total Bytes Now :  3508
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\fresh\index.js
Total Bytes Now :  2711
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\fresh\package.json
Total Bytes Now :  5119
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\http\package.json
Total Bytes Now :  1082
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\http-errors\index.js
Total Bytes Now :  5878
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\http-errors\package.json
Total Bytes Now :  8409
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\iconv-lite\encodings\dbcs-codec.js
Total Bytes Now :  21415
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\iconv-lite\encodings\dbcs-data.js
Total Bytes Now :  29706
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\iconv-lite\encodings\index.js
Total Bytes Now :  30416
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\iconv-lite\encodings\internal.js
Total Bytes Now :  36531
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\iconv-lite\encodings\sbcs-codec.js
Total Bytes Now :  38722
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\iconv-lite\encodings\sbcs-data-generated.js
Total Bytes Now :  70756
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\iconv-lite\encodings\sbcs-data.js
Total Bytes Now :  75442
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\iconv-lite\encodings\tables\big5-added.json
Total Bytes Now :  17717
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\iconv-lite\encodings\tables\cp936.json
Total Bytes Now :  65037
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\iconv-lite\encodings\tables\cp949.json
Total Bytes Now :  103159
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\iconv-lite\encodings\tables\cp950.json
Total Bytes Now :  145515
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\iconv-lite\encodings\tables\eucjp.json
Total Bytes Now :  186579
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\iconv-lite\encodings\tables\gb18030-ranges.json
Total Bytes Now :  188795
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\iconv-lite\encodings\tables\gbk-added.json
Total Bytes Now :  190022
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\iconv-lite\encodings\tables\shiftjis.json
Total Bytes Now :  213804
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\iconv-lite\encodings\utf16.js
Total Bytes Now :  294257
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\iconv-lite\encodings\utf7.js
Total Bytes Now :  303472
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\iconv-lite\lib\bom-handling.js
Total Bytes Now :  1109
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\iconv-lite\lib\extend-node.js
Total Bytes Now :  9810
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\iconv-lite\lib\index.js
Total Bytes Now :  14933
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\iconv-lite\lib\streams.js
Total Bytes Now :  18320
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\iconv-lite\package.json
Total Bytes Now :  323793
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\inherits\inherits.js
Total Bytes Now :  192
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\inherits\inherits_browser.js
Total Bytes Now :  864
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\inherits\package.json
Total Bytes Now :  2372
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\ipaddr.js\ipaddr.min.js
Total Bytes Now :  9738
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\ipaddr.js\lib\ipaddr.js
Total Bytes Now :  19333
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\ipaddr.js\lib\ipaddr.js.d.ts
Total Bytes Now :  22292
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\ipaddr.js\package.json
Total Bytes Now :  33782
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\jsonwebtoken\decode.js
Total Bytes Now :  767
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\jsonwebtoken\index.js
Total Bytes Now :  1043
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\jsonwebtoken\lib\JsonWebTokenError.js
Total Bytes Now :  428
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\jsonwebtoken\lib\NotBeforeError.js
Total Bytes Now :  790
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\jsonwebtoken\lib\psSupported.js
Total Bytes Now :  897
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\jsonwebtoken\lib\timespan.js
Total Bytes Now :  1309
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\jsonwebtoken\lib\TokenExpiredError.js
Total Bytes Now :  1704
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\jsonwebtoken\node_modules\ms\index.js
Total Bytes Now :  3024
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\jsonwebtoken\node_modules\ms\package.json
Total Bytes Now :  4715
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\jsonwebtoken\package.json
Total Bytes Now :  9864
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\jsonwebtoken\sign.js
Total Bytes Now :  16687
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\jsonwebtoken\verify.js
Total Bytes Now :  23626
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\jwa\index.js
Total Bytes Now :  6501
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\jwa\package.json
Total Bytes Now :  8185
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\jws\index.js
Total Bytes Now :  609
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\jws\lib\data-stream.js
Total Bytes Now :  1242
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\jws\lib\sign-stream.js
Total Bytes Now :  3439
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\jws\lib\tostring.js
Total Bytes Now :  3701
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\jws\lib\verify-stream.js
Total Bytes Now :  6927
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\jws\package.json
Total Bytes Now :  9114
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\lodash.includes\index.js
Total Bytes Now :  18716
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\lodash.includes\package.json
Total Bytes Now :  20639
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\lodash.isboolean\index.js
Total Bytes Now :  1819
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\lodash.isboolean\package.json
Total Bytes Now :  3748
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\lodash.isinteger\index.js
Total Bytes Now :  6036
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\lodash.isinteger\package.json
Total Bytes Now :  7971
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\lodash.isnumber\index.js
Total Bytes Now :  2025
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\lodash.isnumber\package.json
Total Bytes Now :  3942
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\lodash.isplainobject\index.js
Total Bytes Now :  3673
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\lodash.isplainobject\package.json
Total Bytes Now :  5656
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\lodash.isstring\index.js
Total Bytes Now :  2302
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\lodash.isstring\package.json
Total Bytes Now :  4219
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\lodash.once\index.js
Total Bytes Now :  7094
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\lodash.once\package.json
Total Bytes Now :  8969
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\media-typer\index.js
Total Bytes Now :  6375
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\media-typer\package.json
Total Bytes Now :  8136
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\merge-descriptors\index.js
Total Bytes Now :  1215
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\merge-descriptors\package.json
Total Bytes Now :  3214
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\methods\index.js
Total Bytes Now :  1040
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\methods\package.json
Total Bytes Now :  3058
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\mime\cli.js
Total Bytes Now :  149
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\mime\mime.js
Total Bytes Now :  2875
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\mime\package.json
Total Bytes Now :  4712
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\mime\src\build.js
Total Bytes Now :  1351
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\mime\src\test.js
Total Bytes Now :  3685
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\mime\types.json
Total Bytes Now :  39952
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\mime-db\db.json
Total Bytes Now :  181417
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\mime-db\index.js
Total Bytes Now :  181553
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\mime-db\package.json
Total Bytes Now :  184322
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\mime-types\index.js
Total Bytes Now :  3663
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\mime-types\package.json
Total Bytes Now :  6001
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\ms\index.js
Total Bytes Now :  2764
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\ms\package.json
Total Bytes Now :  4331
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\negotiator\index.js
Total Bytes Now :  3344
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\negotiator\lib\charset.js
Total Bytes Now :  3081
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\negotiator\lib\encoding.js
Total Bytes Now :  6587
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\negotiator\lib\language.js
Total Bytes Now :  9995
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\negotiator\lib\mediaType.js
Total Bytes Now :  15353
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\negotiator\package.json
Total Bytes Now :  20857
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\object-assign\index.js
Total Bytes Now :  2108
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\object-assign\package.json
Total Bytes Now :  3853
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\on-finished\index.js
Total Bytes Now :  3686
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\on-finished\package.json
Total Bytes Now :  5681
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\parseurl\index.js
Total Bytes Now :  2809
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\parseurl\package.json
Total Bytes Now :  5127
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\passport\lib\authenticator.js
Total Bytes Now :  13429
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\passport\lib\errors\authenticationerror.js
Total Bytes Now :  438
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\passport\lib\framework\connect.js
Total Bytes Now :  1189
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\passport\lib\http\request.js
Total Bytes Now :  2206
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\passport\lib\index.js
Total Bytes Now :  17736
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\passport\lib\middleware\authenticate.js
Total Bytes Now :  13674
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\passport\lib\middleware\initialize.js
Total Bytes Now :  15513
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\passport\lib\sessionmanager.js
Total Bytes Now :  34116
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\passport\lib\strategies\session.js
Total Bytes Now :  2091
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\passport\package.json
Total Bytes Now :  38107
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\passport-http\lib\passport-http\index.js
Total Bytes Now :  249
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\passport-http\lib\passport-http\strategies\basic.js
Total Bytes Now :  3069
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\passport-http\lib\passport-http\strategies\digest.js
Total Bytes Now :  12496
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\passport-http\package.json
Total Bytes Now :  14518
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\passport-jwt\lib\auth_header.js
Total Bytes Now :  305
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\passport-jwt\lib\extract_jwt.js
Total Bytes Now :  4044
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\passport-jwt\lib\helpers\assign.js
Total Bytes Now :  860
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\passport-jwt\lib\index.js
Total Bytes Now :  5079
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\passport-jwt\lib\strategy.js
Total Bytes Now :  10170
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\passport-jwt\lib\verify_jwt.js
Total Bytes Now :  10339
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\passport-jwt\package.json
Total Bytes Now :  12179
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\passport-jwt\test\auth_header-test.js
Total Bytes Now :  824
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\passport-jwt\test\bootstrap.js
Total Bytes Now :  926
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\passport-jwt\test\extrators-test.js
Total Bytes Now :  9132
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\passport-jwt\test\mock_request.js
Total Bytes Now :  9310
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\passport-jwt\test\strategy-init-test.js
Total Bytes Now :  10789
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\passport-jwt\test\strategy-requests-test.js
Total Bytes Now :  14261
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\passport-jwt\test\strategy-validation-test.js
Total Bytes Now :  20308
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\passport-jwt\test\strategy-verify-test.js
Total Bytes Now :  28549
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\passport-jwt\test\testdata.js
Total Bytes Now :  29131
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\passport-strategy\.jshintrc
Total Bytes Now :  305
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\passport-strategy\lib\index.js
Total Bytes Now :  223
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\passport-strategy\lib\strategy.js
Total Bytes Now :  790
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\passport-strategy\package.json
Total Bytes Now :  3028
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\path-to-regexp\index.js
Total Bytes Now :  3328
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\path-to-regexp\package.json
Total Bytes Now :  4832
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\pause\index.js
Total Bytes Now :  595
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\pause\package.json
Total Bytes Now :  1600
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\proxy-addr\index.js
Total Bytes Now :  6000
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\proxy-addr\package.json
Total Bytes Now :  8257
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\qs\dist\qs.js
Total Bytes Now :  24660
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\qs\lib\formats.js
Total Bytes Now :  387
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\qs\lib\index.js
Total Bytes Now :  598
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\qs\lib\parse.js
Total Bytes Now :  9090
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\qs\lib\stringify.js
Total Bytes Now :  17320
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\qs\lib\utils.js
Total Bytes Now :  23524
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\qs\package.json
Total Bytes Now :  50549
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\qs\test\index.js
Total Bytes Now :  81
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\qs\test\parse.js
Total Bytes Now :  27839
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\qs\test\stringify.js
Total Bytes Now :  51912
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\qs\test\utils.js
Total Bytes Now :  57008
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\range-parser\index.js
Total Bytes Now :  2900
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\range-parser\package.json
Total Bytes Now :  5315
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\raw-body\index.js
Total Bytes Now :  6059
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\raw-body\package.json
Total Bytes Now :  8671
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\safe-buffer\index.js
Total Bytes Now :  1529
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\safe-buffer\package.json
Total Bytes Now :  3172
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\safer-buffer\dangerous.js
Total Bytes Now :  1483
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\safer-buffer\package.json
Total Bytes Now :  3249
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\safer-buffer\safer.js
Total Bytes Now :  5359
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\safer-buffer\tests.js
Total Bytes Now :  21094
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\semver\package.json
Total Bytes Now :  1551
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\semver\semver.js
Total Bytes Now :  40354
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\send\index.js
Total Bytes Now :  23287
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\send\node_modules\ms\index.js
Total Bytes Now :  3034
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\send\node_modules\ms\package.json
Total Bytes Now :  4667
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\send\package.json
Total Bytes Now :  30760
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\serve-static\index.js
Total Bytes Now :  4570
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\serve-static\package.json
Total Bytes Now :  6927
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\setprototypeof\index.js
Total Bytes Now :  384
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\setprototypeof\package.json
Total Bytes Now :  2454
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\setprototypeof\test\index.js
Total Bytes Now :  690
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\statuses\codes.json
Total Bytes Now :  1821
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\statuses\index.js
Total Bytes Now :  3909
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\statuses\package.json
Total Bytes Now :  6465
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\toidentifier\index.js
Total Bytes Now :  490
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\toidentifier\package.json
Total Bytes Now :  2680
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\type-is\index.js
Total Bytes Now :  5562
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\type-is\package.json
Total Bytes Now :  7803
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\unpipe\index.js
Total Bytes Now :  1118
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\unpipe\package.json
Total Bytes Now :  2870
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\utils-merge\index.js
Total Bytes Now :  381
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\utils-merge\package.json
Total Bytes Now :  2057
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\vary\index.js
Total Bytes Now :  2930
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\node_modules\vary\package.json
Total Bytes Now :  5073
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\package-lock.json
Total Bytes Now :  1403578
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\package.json
Total Bytes Now :  1404419
Path of file visiting:  C:\Users\ravisher\Desktop\CSCI3916_HW2\server.js
Total Bytes Now :  1408761
1408761
​
